app/controllers/dams_resource_controller.rb

```diff
@@ -30,6 +30,17 @@ def show
     # get metadata from solr
     @document = get_single_doc_via_search(1, {:q => "id:#{params[:id]}"} )
 
+    # generate facet collection list for collection page only
+    if !@document.nil?
```

Comment:
@lsitu we have a number of instances in the show method where we're checking whether @document is nil or not. what are your thoughts on checking @document immediately after the `get_single_soc_via_search` method and returning if it is nil at that point? then we could remove all the other nil checks in the method.